### ISSUE TITLE
Skipping YouTube ads

### ISSUE THREAD

**Issue by martinlockett** [inProgress-team/react-native-youtube#265]
I have implemented my own control buttons and hidden those in the player. The app works well and supports a playing queue; you can skip forwards and backwards through the queue and pause and play as you would expect.

If I tap my "next video" button whilst a pre-roll ad is playing the desired video is played but the overlay is still shown saying "Visit advertiser" in the top right and "Ad * <progress>" in the bottom left and a progress bar. I don't know where the overlay is coming from but the duration and progress bar are for the selected video, not the ad. If I then wait for the video to finish, the next video is shown and the overlay disappears.

![youtube](https://user-images.githubusercontent.com/3380800/36316121-5e5a6004-1331-11e8-9d74-82ea8e264d53.png)

Is this a bug or does it sound like I'm doing something wrong? Is there a recommended strategy for dealing with pre-roll videos?

**Comment by davidohayon669** [comment]
@martinlockett leave me some Video IDs that have pre-rolls and I'll test this behavior. I have yet to experience a pre-roll and also AFAIK Vevo videos are not playable in this component so maybe it's something new that is about to be fixed in YouTube's iFrame

**Comment by martinlockett** [comment]
@davidohayon669 Thanks for the response. Firstly, TroyeSivanVEVO is the YouTube channel name in the above example. As we provide videoIds and not URLs to this component I am not aware of the video being of a special type.

The adverts I'm referring to are 15 second adverts that appear before the video that we have requested. This is not repeatable each time so I do not believe the adverts are associated with specific videoIds. Perhaps my understanding of pre-roll is incorrect or I have used the wrong terminology. To significantly increase the chances of seeing the adverts in an app having this component, delete your app, clear the IDFA (on iOS this is done via Settings -> Privacy -> Advertising), re-install your app and play a video by providing a videoId to the component.
